### PR TITLE
Feature/nu uart icomchannel

### PR DESCRIPTION
**Pull request by Niuslar**
This is a working prototype for the new CUartCom class. It receives data using interrupts and adds it to the queue. It's not pretty and might be overcomplicated, but it works.

### CODE REVIEW

File: Lib/Inc/CUartCom.h [hunk 1]
```diff
@@ -11,25 +11,41 @@
...
 {
-private:
-    UART_HandleTypeDef *mp_huart;
-    GPIO_TypeDef *m_uart_de_port = nullptr;
-    uint16_t m_uart_de_pin;
-
 public:
-    CUartCom(UART_HandleTypeDef *p_huart);
-    CUartCom(UART_HandleTypeDef *p_huart,
+    CUartCom(const std::string name, UART_HandleTypeDef *p_huart);
```
Comment (Vavat) [inline Lib/Inc/CUartCom.h]: The pointer to the hardware is the primary parameter, so should come first.

File: Lib/Inc/CUartCom.h [hunk 2]
```diff
@@ -11,25 +11,41 @@
...
+    CUartCom(std::string name,
+             UART_HandleTypeDef *p_huart,
              GPIO_TypeDef *uart_de_port,
```
Comment (Vavat) [inline Lib/Inc/CUartCom.h:29]: There is a CGpioWrapper class, so you should make use of it as much as possible. Code reuse is a useful thing to do.

File: Lib/Inc/IComChannel.h [hunk 3]
```diff
@@ -22,7 +22,7 @@ class IComChannel
 
     virtual bool isCommandAvailable() = 0;
     virtual std::string getCommand() = 0;
-    virtual void send(std::string message) = 0;
+    virtual void send(const std::string &message) = 0;
```
Comment (Vavat) [inline Lib/Inc/IComChannel.h:30]: Why is it `const` and passed by reference?

File: Lib/Src/CLog.cpp [hunk 4]
```diff
@@ -47,30 +47,30 @@ void CLog::log(log_level_t log_level, const std::string &message)
             "[ERROR]->" + m_module_name + ": " + message + "\n"));
 
         // Send message using UART
-        mp_error_uart->sendMessage(full_msg);
+        mp_error_uart->send(full_msg);
```
Comment (Vavat) [inline Lib/Src/CLog.cpp:56]: Again, you are modifying too many things in one PR. It gets confusing what the purpose of this branch is.

Comment (Vavat) [reply Lib/Src/CLog.cpp:56]: ah... this is because the uart method changed...

File: Lib/Src/CAdcData.cpp [hunk 5]
```diff
@@ -14,7 +14,7 @@
 #define ADC_START_BIT        2
 #define ADC_CONVERSION_COEFF ((float)ADC_VDDA / ADC_RES)
 
-CUartCom uart_adc(&huart2);
+CUartCom uart_adc("uart_adc", &huart2);
```
Comment (Vavat) [inline Lib/Src/CAdcData.cpp]: Why does ADC need a uart?

File: Lib/Inc/IComChannel.h (hunk 3 above)
Comment (Niuslar) [reply Lib/Inc/IComChannel.h:30]: Because we are not changing the string and it is passed by reference to avoid copying a string.

**Comment by Niuslar** [comment]
Created a new PR for the new MCU.

**Pull request closed without merging**